C-Plus-Plus/sort/three_way_merge_sort.cpp

```diff
@@ -0,0 +1,171 @@
+// C++ program to implement Three Way Merge Sort
+
+// Time Complexity: O(n*logn)
+// Space Complexity: O(n)
+
+#include <bits/stdc++.h>
+using namespace std;
+
+
+// Merge the three sections in ascending order
+void merge(int arr[], int beg, int midA, int midB, int end, int temp[])  
+{  
+    int i, j, k, l;
+    i = beg;
+    j = midA; 
+    k = midB; 
+    l = beg;  
+  
+    // Find the smallest element among the three sections 
+    while ((i < midA) && (j < midB) && (k < end))  
+    {  
+        if(arr[i] < arr[j]) 
+        { 
```

Comment:
Use meaningful variable names

Reply:
Here 'beg' is the beginning index of the first section and midA, midB are the mid points that together with the index 'end' divides the whole array into three sections. Isn't the variable names justified?